### ISSUE TITLE
Error in network file: one or more files were not recognized and could not be added

### ISSUE THREAD

**Issue by aayushtewari01** [matsim-org/matsim-code-examples#660]
What are the basic syntax which we should follow in creating a network.
I created a node network in xml format where node id, x coordinate , y coordinate are only present. I have not yet created link id details.
But I want to visualize the nodes in Simunto Via. I got an error "one or more files were not recognized and could not be added"
I am attaching an image of my network file over here (as xml file cannot be attached) .Please help as this is the very first step I had taken after several data processing and would like to see whether the node network is effective or not.

![image](https://user-images.githubusercontent.com/81128541/144587681-f3e8187d-5a1b-4a14-a8f4-556191c2a357.png)

**Comment by Janekdererste** [comment]
We have some [example scenarios](https://github.com/matsim-org/matsim-code-examples/tree/13.x/scenarios) in this very repository. They also contain network files which you could investigate. 

That said, I would recommend to create your networks programmatically. This way the MATSim-SDK takes care of file layout for you. Examples are also in this repository:

- [Craete network from scratch](https://github.com/matsim-org/matsim-code-examples/blob/13.x/src/main/java/org/matsim/codeexamples/network/RunCreateNetworkExample.java)
- [Convert OSM-Data to MATSim-Network](https://github.com/matsim-org/matsim-code-examples/blob/13.x/src/main/java/org/matsim/codeexamples/network/RunCreateNetworkFromOSM.java)

**Comment by kainagel** [comment]
I think that in your case you would need to add an empty links section into your file.  I.e.
```
...
</node>
<links>
</links>
</network>
```

**Comment by aayushtewari01** [comment]
@Janekdererste Thank you Sir for the code links. I have two linked questions with this
1. This code is for 6 number of codes. So If I have to create a network of a million nodes how should I proceed. I have a Digital Road Network in WKT format of Japan and I want to convert it to MATSim input file.?
2. I wanted to use Python  geopandas library to convert my raw data to the type of network data I found in 'equil'example. In my question I posted snapshot using the same method and I do not know where I went wrong.

using text editor I added sample links for few nodes and again tried importing it in Simunto Via but the error remains. Can you please guide. Thanks

**Comment by aayushtewari01** [comment]
@kainagel Sir thanks a lot sir for your comment over here. Sir, I used a raw Digital Road Network of Japan in WKT format using geopandas library of python. Further, I edited the text file manually and added links for few nodes as mentioned by you Sir, even after adding a set of empty links the error remains while uploading the created network in Simunto Via. Error is "One or more files were not recognized and could not be added". 
Would you please further guide. Thank you

**Comment by Janekdererste** [comment]
> even after adding a set of empty links the error remains 

@kainagel meant to add an empty links section. So, in the exmaple you've provided, add 
```
<links></links>
```
after your ```</node>``` tag. This means you have an empty list of links in your network. 

> using text editor I added sample links for few nodes and again tried importing it in Simunto Via but the error remains

You could take one of the network files from our sample scenarios mentioned above, make sure they can be read by Via and then replace the content with your data while maintaining the structure of the file. You could even keep the existing data as a reference and add your own nodes while trying to make things look like our example. Maybe you can spot the differences. Appart from the missing ```<links></links>``` section you are also missing a ```DTD```. Again, this is all inlcuded in the exmple files. 

> This code is for 6 number of codes. So If I have to create a network of a million nodes how should I proceed. I have a Digital Road Network in WKT format of Japan and I want to convert it to MATSim input file.?

Assuming your road network is represented as a bunch of linestrings. You could do something along the lines of the following pseudo code:

```
var nodeList = []
var linkList = []
for (linestring : linestringCollection) 
    for (coordinate : linestring.coordinates)
        var node  = createNode(coordinate) // this method must also assign an id
        var prevNode = getPrevNode() // you would have to handle edge cases like the first coordinate and stuff here

        // create the link here, assuming you get things like speed and capacity from your linestring attributes
        var link = createLink(prevNode, node, linestring.attributes) 
        nodeList.push(node)
        linkList.push(link)

// write the xml yourself including xml header, dtd and the actual data. 
writeXmlNetwork(nodeList, linkList)
```
You have to figure out how to make sure that nodes which are referenced by multiple line strings are not created twice but shared.

**Comment by aayushtewari01** [comment]
@Janekdererste @kainagel Thanks Sir for guidance. So I added empty links and also added the DTD of MATSim Equil example network. I also validated the xml and it was smooth. I could access the sample nodes on Simunto Via. Thanks. I will try to develop the links on Python. I wish I had started with Java since the availability of code in repository is great. But I started learning Python and would like to see how far does it takes me. I will seek further guidance . Thanks
